jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1172,19 +1177,18 @@ public synchronized void start() {
 	 *
 	 * Release the lock before calling threadgroup's remove method for this thread.
 	/*[ENDIF]*/
+	synchronized (lock) {
+		if (started) {
+			/*[MSG "K0341", "Thread is already started"]*/
+			throw new IllegalThreadStateException(com.ibm.oti.util.Msg.getString("K0341")); //$NON-NLS-1$
+		}
+		/*[PR 115667, CMVC 94448] In 1.5, thread is added to ThreadGroup when started */
+		group.add(this);
+	}
 
 	try {
-		synchronized(lock) {
-			if (started) {
-				/*[MSG "K0341", "Thread is already started"]*/
-				throw new IllegalThreadStateException(com.ibm.oti.util.Msg.getString("K0341")); //$NON-NLS-1$
-			}
-
-			/*[PR 115667, CMVC 94448] In 1.5, thread is added to ThreadGroup when started */
-			group.add(this);
-
+		synchronized (lock) {
 			startImpl();
-
 			success = true;
 		}
 	} finally {
```

Comment:
If `!success`, the thread should be removed from the group (independent of `started`).

Reply:
`!started` is to prevent the deadlock as per the comment
```
	 * Deadlock happens when a thread is added to the ThreadGroup
	 * and its start method fails and calls its ThreadGroup's remove method for this thread to be removed while holding this thread's lock.
	 * If there is another call which is holding ThreadGroup's child lock and waiting for this thread's lock,
	 * then deadlock occurs since ThreadGroup's remove method is waiting for its child lock.
	 * Release the lock before calling threadgroup's remove method for this thread.
```

Reply:
It should be removed even if `started` is true (but `success` is false). It wasn't _successfully_ started and should not be left as a member of the thread group.

Reply:
Removed the check of `started` for failed thread removal.